src/lib/ui/DeviceDetails/DeviceDetails.tsx

```diff
@@ -54,6 +57,36 @@ function LockDeviceDetails(props: { device: LockDevice; onBack?: () => void }) {
   const accessCodeLength =
     device.properties?.schlage_metadata?.access_code_length
 
+  function generateDeviceAlerts() {
+    if (device.errors.length === 0 && device.warnings.length === 0) {
```

Comment:
I prefer a more generic check like `if (!devices.errors)` but GitHub Actions is complaining on the basis of the type. Is `devices.errors` and `devices.warnings` always defined and always an array?

(Same goes for the conditions on lines 67 & 76)